src/librustc_session/options.rs

```diff
@@ -920,9 +920,6 @@ options! {DebuggingOptions, DebuggingSetter, basic_debugging_options,
         "enable origins tracking in MemorySanitizer"),
     sanitizer_recover: Vec<Sanitizer> = (vec![], parse_sanitizer_list, [TRACKED],
         "enable recovery for selected sanitizers"),
-    saturating_float_casts: bool = (false, parse_bool, [TRACKED],
-        "make float->int casts UB-free: numbers outside the integer type's range are clipped to \
-        the max/min integer respectively, and NaN is mapped to 0 (default: no)"),
```

Comment:
I would suggest to retain this option (with default true and ability to write `-Z saturating-float-casts=no`) for a while, to make it easier to diagnose performance problems related to this change.

Reply:
Makes sense. Updated.

Reply:
It's probably necessary to make this an `Option<bool>` and use `unwrap_or(true)` to make this work.

Reply:
Hm, have we really not fixed that yet? Seems unfortunate.

Reply:
Ah, sorry, you were right! This did get fixed recently in #70729.

Reply:
I'll leave it as is since I've changed it now and it doesn't matter really (we'll drop this option anyway).